### DISCUSSION TITLE
In Chrome, focus does not consistently start at top of page

### DISCUSSION THREAD

**Discussion by RD-Brown** [nvaccess/nvda#15130] [Q&A]
TestNav is Pearson’s online test delivery platform and is based on Chromium. We typically set the JAWS focus to the top of the page every time a user navigates between items  in a test. However, in Chrome versions higher than 110 we have noticed that the focus does not consistently start at the top of the page, and the Up/Down Arrow behavior doesn’t seem to look at the focus element. (Note that this behavior is also present in JAWS.)

The following code is used to set focus:
util.addFTListener = function () {
    $(document).off('inform.loadingHidden', util.focusFirstTabbable).one('inform.loadingHidden', util.focusFirstTabbable);
};

util.focusFirstTabbable = function () {
    setTimeout(function () { $('#firstTabbable').focus() }, 10);
    util.removeFTListener();
};

util.removeFTListener = function () {
    $(document).off('inform.loadingHidden', util.focusFirstTabbable);
}

### Steps to reproduce:
1.	In Chrome, go to https://test-release.testnav.com/client/index.html
2.	In the Username field enter 1688667634
3.	In the password field enter tc101
4.	Press the Sign In button
5.	On the Test Start screen, press the Start button
6.	On the Terms and Conditions screen, check the “I agree to the terms and conditions of use” checkbox, and press the Continue button.
7.	Press the Start button.
8.	On the Question 1 screen, dismiss the tips modal by pressing the Close button
9.	Begin pressing the Next button, and note that focus is not consistently reset to the top of the page (In our testing, this behavior starts with the initial Test Start page, and continues throughout the form.)

# Actual behavior:

Focus is often placed in the middle of the page content. 
### Expected behavior:

Focus should begin at the top of the page.
### System configuration
Windows 11 Pro, Version    10.0.22000 Build 22000
64-bit, RAM 32.0 GB
Processor    Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz   2.11 GHz

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2023.1 (2023.1.0.27913) 
#### Windows version:
Windows 11
#### Name and version of other software in use when reproducing the issue:
Chrome 114.0.5735.135
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by seanbudd** [comment]
Welcome @RD-Brown ,

We are unable to process this issue as it stands.
To be able to reproduce this issue, we need a minimal code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).

Kind Regards,
NV Access Software Developers

**Comment by michaelDCurran** [comment]
I can reproduce this, but 
as this is an issue with multiple screen readers, I shall close this and convert to a discussion. The issue is either with Chromium or with the Javascript.
Clearly Chromium has either stopped firing the focus event, or the element focus is fired upon is now invalid or does not live long enough or something.
I would suggest watching the focus events in Chromium's accessibility inspector and or using a tool such as AccEvent from Microsoft to verify that the focus event is being fired by Chromium.
Perhaps also try delaying the focus event by 50 ms or so.
